### PR TITLE
update documentation

### PR DESCRIPTION
**Pull request by shenlebantongying**
* use repology.org to replace manually created linux package list [0]
* update debian/ubuntu dependencies (delete outdated fedora one)
* update all build doc to qt6
* add cmake flags info
* merge all build instructions into one
* add more ocr examples
* add links and fix dictionary formats info
* misc

[0]

The repology's list will auto-expand https://repology.org/project/goldendict-ng/versions

image

### CODE REVIEW

File: website/docs/dictformats.md [hunk 1]
```diff
@@ -2,13 +2,13 @@ Popular dictionary formats are all supported.
 
 ## Local Dictionaries Sources
 
-* MDict dictionaries (.mdx/.mdd)
-* StarDict dictionaries (.ifo/.dict./.idx/.syn)
+* [MDict](https://www.mdict.cn/) dictionaries (.mdx/.mdd)
+* [StarDict](http://www.huzheng.org/stardict/) dictionaries (.ifo/.dict./.idx/.syn)
 * DSL dictionaries (ABBYY Lingvo source files .dsl(.dz))
```
Comment (xiaoyifang) [inline website/docs/dictformats.md:7]: https://anatoly314.github.io/dsl-manual/#../index.html
dsl document

Comment (shenlebantongying) [reply website/docs/dictformats.md:7]: It is not official, and the actual source is 

https://lingvoboard.ru/store/html/DSLReference_HTML/index.html

:sweat_smile:

**Comment by shenlebantongying** [comment]
If I understand correctly, the "First word only" for capture2text is an optional step for only "Forward text line OCR".

File: website/docs/dictformats.md (hunk 1 above)
Comment (shenlebantongying) [reply website/docs/dictformats.md]: Added

**Comment by shenlebantongying** [comment]
The Debian dependencies on doc is updated according to https://salsa.debian.org/debian/goldendict-ng/-/blob/master/debian/control

The `libao-dev` & `libtiff-dev` are no longer needed. @hosiet `:)`

File: website/docs/dictformats.md (hunk 1 above)
Comment (shenlebantongying) [reply website/docs/dictformats.md]: Added.

**Pull request merged**